src/handshakes/abstract-handshake.ts

```diff
@@ -1,10 +1,11 @@
 import { equals as uint8ArrayEquals } from 'uint8arrays/equals'
 import { concat as uint8ArrayConcat } from 'uint8arrays/concat'
 import { fromString as uint8ArrayFromString } from 'uint8arrays'
-import type { bytes, bytes32, uint64 } from '../@types/basic.js'
+import type { bytes, bytes32 } from '../@types/basic.js'
 import type { CipherState, MessageBuffer, SymmetricState } from '../@types/handshake.js'
 import type { ICryptoInterface } from '../crypto.js'
 import { logger } from '../logger.js'
+import { Nonce } from '../nonce.js'
 
 export const MIN_NONCE = 0
 // For performance reasons, the nonce is represented as a JS `number`
```

Comment:
It may be worth revisiting this comment now that we have a more performant Nonce implementation.

Can you try changing `Nonce.n` to a `bigint`, changing MAX_NONCE to `2 ** 64 - 2`, and the dataview set to `setBigUint64`?
If it's not extremely expensive we should do it, since its part of the noise spec.

Reply:
ouch, yea lets just stick with `number`

I think there's a lingering bug, where the MAX_NONCE doesn't match the overflow from `dataview.setUint32`.
We might either change MAX_NONCE or do setUint32 twice for the high and low bytes.